backend/session/logind.c

```diff
@@ -70,22 +69,24 @@ static int logind_take_device(struct wlr_session *base, const char *path) {
 		session->path, "org.freedesktop.login1.Session", "TakeDevice",
 		&error, &msg, "uu", major(st.st_rdev), minor(st.st_rdev));
 	if (ret < 0) {
-		wlr_log(WLR_ERROR, "Failed to take device '%s': %s", path, error.message);
+		wlr_log(WLR_ERROR, "Failed to take device '%s': %s", path,
+			error.message);
 		goto error;
 	}
 
+	int fd = -1;
 	int paused = 0;
 	ret = sd_bus_message_read(msg, "hb", &fd, &paused);
 	if (ret < 0) {
 		wlr_log(WLR_ERROR, "Failed to parse D-Bus response for '%s': %s",
-			path, strerror(-ret));
+			path, error.message);
```

Comment:
```error.message``` is only set by ```sd_bus_call_method```. Why would you use it here?

Reply:
Good catch